src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -76,316 +68,67 @@ public enum Flags
+        // Usually our tokens are less than 128 characters long, unless it's a large string.
+        private const int DesiredStringBuilderSize = 128;
 
-            _keywords = new Dictionary<string, TokKind>();
-            _punctuators = new Dictionary<string, TokKind>();
-            _cache = null;
-
-            // Punctuators
-            AddPunctuator(PunctuatorOr, TokKind.Or);
-            AddPunctuator(PunctuatorAnd, TokKind.And);
-            AddPunctuator(PunctuatorBang, TokKind.Bang);
-            AddPunctuator(PunctuatorAdd, TokKind.Add);
-            AddPunctuator(PunctuatorSub, TokKind.Sub);
-            AddPunctuator(PunctuatorMul, TokKind.Mul);
-            AddPunctuator(PunctuatorDiv, TokKind.Div);
-            AddPunctuator(PunctuatorCaret, TokKind.Caret);
-            AddPunctuator(PunctuatorParenOpen, TokKind.ParenOpen);
-            AddPunctuator(PunctuatorParenClose, TokKind.ParenClose);
-            AddPunctuator(PunctuatorEqual, TokKind.Equ);
-            AddPunctuator(PunctuatorLess, TokKind.Lss);
-            AddPunctuator(PunctuatorLessOrEqual, TokKind.LssEqu);
-            AddPunctuator(PunctuatorGreater, TokKind.Grt);
-            AddPunctuator(PunctuatorGreaterOrEqual, TokKind.GrtEqu);
-            AddPunctuator(PunctuatorNotEqual, TokKind.LssGrt);
-            AddPunctuator(LocalizedPunctuatorListSeparator, TokKind.Comma);
-            AddPunctuator(PunctuatorDot, TokKind.Dot);
-            AddPunctuator(PunctuatorColon, TokKind.Colon);
-            AddPunctuator(PunctuatorCurlyOpen, TokKind.CurlyOpen);
-            AddPunctuator(PunctuatorCurlyClose, TokKind.CurlyClose);
-            AddPunctuator(PunctuatorBracketOpen, TokKind.BracketOpen);
-            AddPunctuator(PunctuatorBracketClose, TokKind.BracketClose);
-            AddPunctuator(PunctuatorAmpersand, TokKind.Ampersand);
-            AddPunctuator(PunctuatorPercent, TokKind.PercentSign);
-            AddPunctuator(LocalizedPunctuatorChainingSeparator, TokKind.Semicolon);
-            AddPunctuator(PunctuatorAt, TokKind.At);
-
-            //Commenting punctuators
-            AddPunctuator(PunctuatorBlockComment, TokKind.Comment);
-            AddPunctuator(PunctuatorLineComment, TokKind.Comment);
-
-            // Keywords
-            AddKeyword(KeywordTrue, TokKind.True);
-            AddKeyword(KeywordFalse, TokKind.False);
-            AddKeyword(KeywordIn, TokKind.In);
-            AddKeyword(KeywordExactin, TokKind.Exactin);
-            AddKeyword(KeywordSelf, TokKind.Self);
-            AddKeyword(KeywordParent, TokKind.Parent);
-            AddKeyword(KeywordAnd, TokKind.KeyAnd);
-            AddKeyword(KeywordOr, TokKind.KeyOr);
-            AddKeyword(KeywordNot, TokKind.KeyNot);
-            AddKeyword(KeywordAs, TokKind.As);
-
-            PopulateKeywordArrays();
-        }
+        private static readonly Lazy<TexlLexer> _invariantDecimalSeparatorLexer = new Lazy<TexlLexer>(() => new TexlLexer(PunctuatorDecimalSeparatorInvariant));
```

Comment:
>Lazy [](http://example.com/codeflow?start=32&length=4)

can we avoid using Lazy<T>?  It's an extra level of complexity and We already inevitably create these, so lazy<t> doesn't buy us much. 
And C#'s static ctor fields will already accomplish something very similar.